server/tests/stt_ntb_ninjs_parse_test.py

```diff
@@ -0,0 +1,226 @@
+                self.item = self.items[0]
+            else:
+                self.item = {}
+
+    def test_can_parse_text_item(self):
+        """Test that parser can parse text items."""
+        assert self.item["type"] == "text"
+
+    def test_datetime_without_timezone(self):
+        """Test datetime parsing without timezone (assumes Helsinki)."""
+        assert self.item["type"] == "text"
+
+    def test_sanitise_stt_tt_html(self):
+        """Test HTML sanitization."""
+        result = self.item["body_html"]
+
+        assert "<html>" not in result
+        assert "<body>" not in result
+
+        assert "Norway" in result and "food prices" in result
+
+    def test_strip_ignored_fields(self):
+        """Test that ignored fields are stripped."""
+        assert "place" not in self.item
+        assert "genre" not in self.item
+
+    def test_abstract_prepending(self):
+        """Test that description_html is prepended to body_html."""
+        assert (
+            "Norway" in self.item["body_html"]
+            and "food prices" in self.item["body_html"]
+        )
+        assert "NorgesGruppen" in self.item["body_html"]
+
+    def test_cv_lookup_function(self):
+        """Test CV lookup functionality."""
+        cv_items = [
+            {"qcode": "20000023", "name": "country music"},
+            {"qcode": "20000021", "name": "music genre"},
+        ]
+
+        result = _cv_lookup(cv_items, "20000023")
+        assert result is not None
+        assert result["qcode"] == "20000023"
+
+        assert _cv_lookup(cv_items, "99999999") is None
+
+        assert _cv_lookup(cv_items, "") is None
+
+    def test_fixture_structure(self):
+        """Test that the fixture file has the expected structure and mappings."""
+        # Basic structure from parsed item
+        assert self.item["type"] == "text"
+        assert (
+            self.item["headline"]
+            == "Food prices rise again as grocery giant expands into pharmacies"
+        )
+        assert self.item["byline"] == "Jecaterina Mantsinen"
+        assert self.item["slugline"] == "test-imatrics"
+
+        # Test exact qcodes produced by parser - any deviation indicates a parser bug
+        subject_qcodes = {
+            s.get("qcode") for s in self.item.get("subject", []) if s.get("qcode")
+        }
+        anpa_qcodes = {
+            a.get("qcode") for a in self.item.get("anpa_category", []) if a.get("qcode")
+        }
+
+        assert subject_qcodes == {
+            "20000023",
+            "20000021",
+            "20000018",
+            "20000002",
+            "01000000",
+            "06000000",
+            "01011000",
+        }
+
+        # Allow environments:
+        #  - without vocab: empty set()
+        #  - with vocab-only mapping: {"14"}
+        #  - with fallback only: {"12"}
+        #  - with both mapped and fallback present: {"14", "12"}
+        assert anpa_qcodes in (set(), {"14"}, {"12"}, {"14", "12"})
```

Comment:
here I suppose it should be only one of those 4 options, but it's hard to say which one